ddtrace/internal/_encoding.pyx

```diff
@@ -273,37 +273,60 @@ cdef class Packer(object):
 
         return ret
 
-    cdef int _pack(self, object o) except -1:
+    cdef _pack_trace(self, object trace):
         cdef int ret
         cdef Py_ssize_t L
 
-        if o is None:
+        if trace is None:
```

Comment:
if we make the change above, can we assume that this is always a list?

Reply:
if we cannot use `list trace`, then I would suggest moving the `PyList_CheckExact(trace)` here and raise an exception if we don't have a list. Encoding nil is weird.